### PR TITLE
Added LangChain handler to the examples.

### PR DESCRIPTION
**Pull request by cam-barts**
Hi there! 👋🏼 

I've been using NiceGUI quite a bit lately to build some experiments and demos at work to help demystify AI stuff to my colleagues. I built this handler to show what was going on in the chains and I frequently copy paste it to new demos and the like, and in the spirit of [Hacktoberfest](https://hacktoberfest.com/) 🎃 I thought I'd try to give back in the form of a contribution. 

This doesn't fit very squarely into the NiceGUI code itself, as it's really just to help interface with LangChain, so I didn't update any actual NiceGUI source at all, but I did add it to the existing LLM example. 

I originally modeled it after the [Streamlit handler](https://github.com/langchain-ai/langchain/blob/f4bec9686d6a2d07a0f0540b34f9313b0cd5ae89/libs/langchain/langchain/callbacks/streamlit/streamlit_callback_handler.py#L225C4-L225C4).

I'll also be creating a PR to add the handler directly to LangChain similarly to the Streamlit handler, but I wanted this bit of code to be accessible for anyone who might find it useful and were going through the examples (which are excellent btw).

### CODE REVIEW

File: examples/chat_with_ai/main.py [hunk 1]
```diff
@@ -5,10 +5,13 @@
 from langchain.chat_models import ChatOpenAI
 
 from nicegui import Client, ui
+from log_callback_handler import NiceGuiLogElementCallbackHandler
 
-OPENAI_API_KEY = 'not-set'  # TODO: set your OpenAI API key here
+OPENAI_API_KEY = "not-set"  # TODO: set your OpenAI API key here
```
Comment (rodja) [inline examples/chat_with_ai/main.py:10]: Please keep using single quotes in this example. The whole codebase is using it like this and we like consistency.

File: examples/chat_with_ai/main.py [hunk 2]
```diff
@@ -5,10 +5,13 @@
...
 
-llm = ConversationChain(llm=ChatOpenAI(model_name='gpt-3.5-turbo', openai_api_key=OPENAI_API_KEY))
+llm = ConversationChain(
+    llm=ChatOpenAI(model_name="gpt-3.5-turbo", openai_api_key=OPENAI_API_KEY),
```
Comment (rodja) [inline examples/chat_with_ai/main.py:13]: Please do not reformat existing code if it does not change. We use a line length of 120.

**Review by rodja — CHANGES_REQUESTED** [review]

**Comment by rodja** [comment]
Thanks for the PR @cam-barts. It's a bit difficult to identify your actual changes because you also applied different formatting. As far as I understand it, the log is only useful when using an agent like in the Streamlit example: https://python.langchain.com/docs/integrations/callbacks/streamlit. This example uses gpt-3.5-turbo and I have not been able to see any logs. Maybe it should be a separate example?

**Comment by cam-barts** [comment]
@rodja Thanks for your patience with the linter issues! I've pushed a new change that fixes all of those, so what's left should just be the addition of the handler. 

The log panel is useful for showing what the chain is doing. In this example, the chain is fairly simple so the output isn't super complex. I put the log on a different tab, so the actual chat interface isn't muddied up a lot. 

Here's a screenshot of the output when I ran this example locally:

Pasted image 20231019115548

A callback handler isn't only useful in agents, and indeed the [callback handler docs](https://python.langchain.com/docs/modules/callbacks/#get-started) use a basic chain as an example. 

This panel becomes more useful as you build more complex chains and agents, in which the agent or chain can choose tools, make decisions based on previous outputs, and check itself. A more complex non-agent chain that I've used it for has been the SQLDatabaseChain with sample rows and custom table data. 

It may not be especially useful in final customer facing applications. In my experience however, it's proven invaluable for the explainability of LangChain and llms in general, especially when you are throwing together quick and dirty deployable PoCs where everyone might not have access to the console logs.

**Comment by rodja** [comment]
I have not seen any logs in the second tab on my first try. I just tested again. But there seems to be a problem with the API server which uncovers the missing callback `on_retry`:

```
Retrying langchain.chat_models.openai.acompletion_with_retry.<locals>._completion_with_retry in 4.0 seconds as it raised APIError: Internal server error {
    "error": {
        "message": "Internal server error",
        "type": "auth_subrequest_error",
        "param": null,
        "code": "internal_error"
    }
}
 500 {'error': {'message': 'Internal server error', 'type': 'auth_subrequest_error', 'param': None, 'code': 'internal_error'}} <CIMultiDictProxy('Date': 'Fri, 20 Oct 2023 03:14:12 GMT', 'Content-Type': 'application/json; charset=utf-8', 'Content-Length': '166', 'Connection': 'keep-alive', 'Vary': 'Origin', 'x-request-id': 'f2b01a4ccacaac33dc1e4844a3a1e482', 'strict-transport-security': 'max-age=15724800; includeSubDomains', 'CF-Cache-Status': 'DYNAMIC', 'Server': 'cloudflare', 'CF-RAY': '818e16be1faf18c9-FRA', 'alt-svc': 'h3=":443"; ma=86400')>.
Error in NiceGuiLogElementCallbackHandler.on_retry callback: 'NiceGuiLogElementCallbackHandler' object has no attribute 'on_retry'
```

**Comment by rodja** [comment]
Now the API error is gone. But I do not see any log output when doing my first request. Only on the second it shows up:

Oct-20-2023 05-40-00

**Comment by cam-barts** [comment]
@rodja 
Great catch! I tinkered around a bit and found that adding an explicit `update()` call after the push gets it to render after the first message. I pushed that up as an addition to this PR.

**Review by rodja — APPROVED** [review]
Ok. I've modified the layout a bit and got rid of awaiting the client connection. Ready to merge.

**Comment by cam-barts** [comment]
@rodja Thank you so much for merging my PR! 🙌 If it aligns with the project's policies and you're comfortable with it, I'd appreciate it if you could consider adding the 'hacktoberfest-accepted' label to this PR. It would help me count this contribution towards my [Hacktoberfest](https://hacktoberfest.com/) goals. If not, no worries at all—your initial feedback and merging of the PR are greatly appreciated! 😊I'll also reference this PR when I make the PR to LangChain to try to get the handler pulled in.

**Pull request merged**